src/main/java/org/mskcc/cmo/messaging/impl/NATSGatewayImpl.java

```diff
@@ -46,16 +42,11 @@ public void publish(String topic, Object message) throws Exception {
     }
 
     @Override
-    public void subscribe(String topic, Class messageClass) throws Exception {
+    public void subscribe(String topic, MessageHandler messageHandler) throws Exception {
     	if (stanConnection == null) return;
         if (!subscribers.containsKey(topic)) {
-        	
-        	Subscription sub = stanConnection.subscribe(topic, new MessageHandler() {
-        		public void onMessage(Message m) {
-                	String json = new String(m.getData(), StandardCharsets.UTF_8);
-                    Object message = gson.fromJson(json, messageClass);
-                }
-             }, new SubscriptionOptions.Builder().durableName(durableName).build());
+        	Subscription sub = stanConnection.subscribe(topic, messageHandler
+        			, new SubscriptionOptions.Builder().durableName(durableName).build());
```

Comment:
I missed this before - shouldn't this durableName be topic specific?

If so, maybe it gets computed:

durableName = client_id + topic?